flow/scripts/cts.tcl

```diff
@@ -18,21 +18,30 @@ if {[info exist ::env(CTS_CLUSTER_DIAMETER)]} {
   set cluster_diameter 100
 }
 
+proc save_progress {stage} {
+  puts "Run 'make gui_$stage.odb' to load progress snapshot"
+  write_db $::env(RESULTS_DIR)/$stage.odb
+  write_sdc $::env(RESULTS_DIR)/$stage.sdc
+}
+
 if {[info exist ::env(CTS_BUF_DISTANCE)]} {
-clock_tree_synthesis -root_buf "$::env(CTS_BUF_CELL)" -buf_list "$::env(CTS_BUF_CELL)" \
-                     -sink_clustering_enable \
-                     -sink_clustering_size $cluster_size \
-                     -sink_clustering_max_diameter $cluster_diameter \
-                     -distance_between_buffers "$::env(CTS_BUF_DISTANCE)" \
-                     -balance_levels
+  clock_tree_synthesis -root_buf "$::env(CTS_BUF_CELL)" -buf_list "$::env(CTS_BUF_CELL)" \
+                      -sink_clustering_enable \
+                      -sink_clustering_size $cluster_size \
+                      -sink_clustering_max_diameter $cluster_diameter \
+                      -distance_between_buffers "$::env(CTS_BUF_DISTANCE)" \
+                      -balance_levels
 } else {
-clock_tree_synthesis -root_buf "$::env(CTS_BUF_CELL)" -buf_list "$::env(CTS_BUF_CELL)" \
-                     -sink_clustering_enable \
-                     -sink_clustering_size $cluster_size \
-                     -sink_clustering_max_diameter $cluster_diameter \
-                     -balance_levels
+  clock_tree_synthesis -root_buf "$::env(CTS_BUF_CELL)" -buf_list "$::env(CTS_BUF_CELL)" \
+                      -sink_clustering_enable \
+                      -sink_clustering_size $cluster_size \
+                      -sink_clustering_max_diameter $cluster_diameter \
+                      -balance_levels
 }
 
+if {[info exist ::env(CTS_SNAPSHOTS)]} {
+  save_progress 4_1_pre_repair_clock_nets
```

Comment:
What information do you get from this?

Reply:
UUhhh... this one I added for good measure. The one below, I can see negative hold slack prior to setup/hold repair.